pandas/core/internals.py

```diff
@@ -1795,6 +1797,99 @@ def _unstack(self, unstacker_func, new_columns):
         return blocks, mask
 
 
+class ExtensionBlock(NonConsolidatableMixIn, Block):
+    """Block for holding extension types.
+
+    Notes
+    -----
+    This is the holds all 3rd-party extension types. It's also the immediate
+    parent class for our internal extension types' blocks, CategoricalBlock.
+
+    ExtensionArrays are limited to 1-D.
+    """
+    is_extension = True
+    is_bool = False
+
+    def __init__(self, values, placement, ndim=None):
+        self._holder = type(values)
```

Comment:
might need some assertions here about values, IOW it must be a sub-class of ExtensionArray?